src/cascadia/Remoting/WindowManager.cpp

```diff
@@ -83,27 +144,234 @@ namespace winrt::Microsoft::Terminal::Remoting::implementation
                                                       CLSCTX_LOCAL_SERVER);
     }
 
+    void WindowManager::_createMonarchAndCallbacks()
+    {
+        _createMonarch();
+        const auto isKing = _areWeTheKing();
+
+        TraceLoggingWrite(g_hRemotingProvider,
+                          "WindowManager_ConnectedToMonarch",
+                          TraceLoggingUInt64(_monarch.GetPID(), "monarchPID", "The PID of the new Monarch"),
+                          TraceLoggingBoolean(isKing, "isKing", "true if we are the new monarch"),
+                          TraceLoggingLevel(WINEVENT_LEVEL_VERBOSE));
+
+        if (_peasant)
+        {
+            // Inform the monarch of the time we were last activated
+            _monarch.HandleActivatePeasant(_peasant.GetLastActivatedArgs());
+        }
+
+        if (!isKing)
+        {
+            return;
+        }
+        // Here, we're the king!
+        //
+        // This is where you should do any additional setup that might need to be
+        // done when we become the king. THis will be called both for the first
+        // window, and when the current monarch dies.
+
+        _monarch.FindTargetWindowRequested({ this, &WindowManager::_raiseFindTargetWindowRequested });
+    }
+
     bool WindowManager::_areWeTheKing()
     {
         const auto kingPID{ _monarch.GetPID() };
         const auto ourPID{ GetCurrentProcessId() };
         return (ourPID == kingPID);
     }
 
-    Remoting::IPeasant WindowManager::_createOurPeasant()
+    Remoting::IPeasant WindowManager::_createOurPeasant(std::optional<uint64_t> givenID)
     {
         auto p = winrt::make_self<Remoting::implementation::Peasant>();
+        if (givenID)
+        {
+            p->AssignID(givenID.value());
+        }
         _peasant = *p;
         _monarch.AddPeasant(_peasant);
 
-        // TODO:projects/5 Spawn a thread to wait on the monarch, and handle the election
+        TraceLoggingWrite(g_hRemotingProvider,
+                          "WindowManager_CreateOurPeasant",
+                          TraceLoggingUInt64(_peasant.GetID(), "peasantID", "The ID of our new peasant"),
+                          TraceLoggingLevel(WINEVENT_LEVEL_VERBOSE));
 
         return _peasant;
     }
 
+    // Method Description:
+    // - Attempt to connect to the monarch process. This might be us!
+    // - For the new monarch, add us to their list of peasants.
+    // Arguments:
+    // - <none>
+    // Return Value:
+    // - true iff we're the new monarch process.
+    bool WindowManager::_performElection()

```

Comment:
I mean, should this be `runningForMonarch`, if the empire is a democratic one. The name `performElection` feels a little off to me.

Reply:
typo: "iff"

Reply:
Ah, but it is [not a typo](https://en.wikipedia.org/wiki/If_and_only_if)!